scripts/duplicate_pr.py

```diff
@@ -0,0 +1,124 @@
+        exit(1)
+    try:
+        git.checkout('-b', new_branch_name)
+    except sh.ErrorReturnCode_128 as e:
+        print(red(e.stderr.decode()))
+        exit(1)
+
+
+def git_fetch_pr_branch(branch_name:str, git=None):
+    git = git or get_git()
+    # fetch remote branch without switching branches
+    input = "{0}:{0}".format(branch_name)
+    try:
+        git.fetch("origin", input)
+    except sh.ErrorReturnCode_1 as e:
+        print(red(e.stderr.decode()))
+        exit(1)
+
+
+def get_new_commits(base_branch: str, curr_branch:str, git=None):
+    git = git or get_git()
+    base_commit = merge_base_commit(base_branch, curr_branch)
+    recent_commit = latest_commit(curr_branch)
+
+    commits_range = "{}..{}".format(base_commit, recent_commit)
+    interested_commits = git("rev-list", "--no-merges", commits_range).split()
+    return interested_commits
+
+
+def cherry_pick_new_commits(commits:list[str], branch:str, git=None):
+    git = git or get_git()
+    git.checkout(branch)
+    for commits in reversed(commits):
+        git("cherry-pick", commits)
+
+
+def git_push_pr(branch:str, git=None):
+    git = git or get_git()
+    git.push("origin", branch)
+
+
+def merge_base_commit(branch1: str, branch2:str, git=None):
+    git = git or get_git()
+    return str(git("merge-base", branch1, branch2).replace("\n", ""))
+
+
+def latest_commit(branch:str, git=None):
+    git = git or get_git()
+    return str(git("rev-parse", branch).replace("\n", ""))
+
+
+def _wrap_with(code):
+
+    def inner(text, bold=False):
+        c = code
+
+        if bold:
+            c = "1;%s" % c
+        return "\033[%sm%s\033[0m" % (c, text)
+    return inner
+
+
+red = _wrap_with('31')
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Duplicate and push PR between formplayer/master')
+    parser.add_argument('orig_source_branch', type=str, help="Branch name of PR to be duplicated")
+    parser.add_argument('orig_target_branch', type=str, help="Name of branch the original PR merged into",
+                            choices = [key.value for key in BranchName])
+    args = parser.parse_args()
+
+    new_source_branch = "copy_of_" + args.orig_source_branch
+    new_target_branch = get_target_branch(args.orig_target_branch)
+
+    print("Creating branch {} from {}".format(new_source_branch, new_target_branch))
+    git_create_branch(orig_branch_name=new_target_branch, new_branch_name=new_source_branch)
+
+    print("Fetching {}".format(args.orig_source_branch))
+    git_fetch_pr_branch(args.orig_source_branch)
+
+    print("Getting new commits from {}".format(args.orig_source_branch))
+    new_commits = get_new_commits(args.orig_target_branch, args.orig_source_branch)
+
+    print("Cherry-picking commits from {} to {}".format(args.orig_source_branch, new_source_branch))
+    cherry_pick_new_commits (new_commits, new_source_branch)
+
+    print("Pushing {}".format(new_source_branch))
+    git_push_pr(new_source_branch)
```

Comment:
Is it also possible to create a new PR after this from the new source branch ? Maybe not but wanted to check.

Reply:
I don't think it's possible with just git - there's an extension that can do it but don't think we should do that.
https://stackoverflow.com/questions/4037928/can-you-issue-pull-requests-from-the-command-line-on-github

I was thinking to create the pull request using github action

Reply:
makes sense, thanks!